src/Microsoft.VisualBasic/tests/ConversionsTests.cs

```diff
@@ -5,83 +5,2952 @@
+            yield return new object[] { (ULongEnum)ulong.MaxValue };
+            yield return new object[] { long.MinValue };
+            yield return new object[] { long.MaxValue };
+            yield return new object[] { (LongEnum)long.MinValue };
+            yield return new object[] { (LongEnum)long.MaxValue };
+            yield return new object[] { float.MinValue };
+            yield return new object[] { float.MaxValue };
+            yield return new object[] { float.PositiveInfinity };
+            yield return new object[] { float.NegativeInfinity };
+            yield return new object[] { float.NaN };
+            yield return new object[] { double.MinValue };
+            yield return new object[] { double.MaxValue };
+            yield return new object[] { double.PositiveInfinity };
+            yield return new object[] { double.NegativeInfinity };
+            yield return new object[] { double.NaN };
+            yield return new object[] { decimal.MinValue };
+            yield return new object[] { decimal.MaxValue };
+            yield return new object[] { "18446744073709551616" };
+            yield return new object[] { "1844674407370955161618446744073709551616" };
+            yield return new object[] { double.PositiveInfinity.ToString() };
+            yield return new object[] { double.NegativeInfinity.ToString() };
+            yield return new object[] { double.NaN.ToString() };
+        }
+
+        [Theory]
+        [MemberData(nameof(ToSByte_OverflowObject_TestData))]
+        public void ToSByte_OverflowObject_ThrowsOverflowException(object value)
+        {
+            Assert.Throws<OverflowException>(() => Conversions.ToSByte(value));
+        }
+
+        public static IEnumerable<object[]> ToUShort_String_TestData()
+        {
+            yield return new object[] { null, ushort.MinValue };
+            yield return new object[] { "0", ushort.MinValue };
+            yield return new object[] { "1", (ushort)1 };
+            yield return new object[] { "&h5", (ushort)5 };
+            yield return new object[] { "&h0", ushort.MinValue };
+            yield return new object[] { "&o5", (ushort)5 };
+            yield return new object[] { " &o5", (ushort)5 };
+            yield return new object[] { "&o0", ushort.MinValue };
+            yield return new object[] { 1.1.ToString(), (ushort)1 };
+        }
+
+        [Theory]
+        [MemberData(nameof(ToUShort_String_TestData))]
+        public void ToUShort_String_ReturnsExpected(string value, ushort expected)
+        {
+            AssertEqual(expected, Conversions.ToUShort(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidString_TestData))]
+        public void ToUShort_InvalidString_ThrowsInvalidCastException(string value)
+        {
+            Assert.Throws<InvalidCastException>(() => Conversions.ToUShort(value));
+        }
+
+        public static IEnumerable<object[]> ToUShort_OverflowString_TestData()
```

Comment:
This is exactly same as ToByte_OverflowString_TestData

Reply:
The data is common among many data types. can we just use one copy everywhere

Reply:
kind of common - yes. fixed